modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/HashAggregateNode.java

```diff
@@ -239,6 +239,13 @@ private Grouping(byte grpId, ImmutableBitSet grpFields) {
 
             handler = context().rowHandler();
 
+            reset();
```

Comment:
my feeling of beauty tells me that it`s not elegant to clean the map on constructor ) can you change it ?

Reply:
I've reworked this a bit. Please take a look